spring/spring-data/src/test/java/io/quarkus/ts/spring/data/rest/additional/OpenAPIIT.java

```diff
@@ -0,0 +1,79 @@
+
+@Tag("QUARKUS-2788")
+@QuarkusScenario
+@DisabledOnNative(reason = "https://github.com/quarkusio/quarkus/issues/30919")
+public class OpenAPIIT extends AbstractDbIT {
+    @Test
+    void rolesAllowedResourceAuthPermitted() {
+        Response response = app.given()
+                .accept(ContentType.JSON)
+                .get("/q/openapi");
+        response.then()
+                .statusCode(HttpStatus.SC_OK)
+                .header("Content-Type", "application/json;charset=UTF-8");
+        assertEquals(HttpStatus.SC_OK, response.statusCode());
+        JsonPath json = response.getBody().jsonPath();
+        assertEquals("spring-data API", json.getString("info.title"));
+
+        assertNotNull(json.getString("paths./article-jpa.get"));
+
+        json.setRootPath("paths./article-jpa.get");
+        assertEquals("page", json.getString("parameters[0].name"));
+        assertEquals("integer", json.getString("parameters[0].schema.type"));
+        assertEquals("size", json.getString("parameters[1].name"));
+        assertEquals("integer", json.getString("parameters[1].schema.type"));
+        assertEquals("sort", json.getString("parameters[2].name"));
+        assertEquals("array", json.getString("parameters[2].schema.type"));
+
+        assertEquals(3, json.getList("parameters").size());
+
+        assertEquals("#/components/schemas/Article",
+                json.getString("responses.200.content.\"application/json\".schema.items.$ref"));
+
+        json.setRootPath("").setRootPath("paths./article-jpa.post");
+        assertEquals("#/components/schemas/Article",
+                json.getString("requestBody.content.\"application/json\".schema.$ref"));
+        assertEquals("#/components/schemas/Article",
+                json.getString("responses.201.content.\"application/json\".schema.$ref"));
+        json.setRootPath("");
+        assertNotNull(json.getString("paths./article-jpa.get"));
+        assertNotNull(json.getString("paths.\"/article-jpa/{id}\".get"));
+
+        List<String> list = json.getList("components.schemas.Article.required");
+        assertEquals(2, list.size());
+        assertThat(list, contains("name", "author"));
+
+        json.setRootPath("").setRootPath("components.schemas.Article.properties");
+        assertEquals("int64", json.getString("id.format"));
+        assertEquals("string", json.getString("name.type"));
+        assertEquals("string", json.getString("author.type"));
+        assertEquals("#/components/schemas/Library", json.getString("library.$ref"));
+
+        json.setRootPath("");
+        assertEquals("2022-03-10T12:15:50", json.getString("components.schemas.LocalDateTime.example"));
+
+        assertEquals("http", json.getString("components.securitySchemes.SecurityScheme.type"));
+        assertEquals("Authentication", json.getString("components.securitySchemes.SecurityScheme.description"));
+        assertEquals("basic", json.getString("components.securitySchemes.SecurityScheme.scheme"));
+
```

Comment:
I'm missing some asserts about secured endpoints' security schemas:
```java
        assertEquals("admin", json.getString("paths.'/secured/deny-all/{id}'.get.security[0].SecurityScheme[0]"));
        assertEquals("admin", json.getString("paths.'/secured/roles-allowed'.get.security[0].SecurityScheme[0]"));
        assertEquals("admin", json.getString("paths.'/secured/roles-allowed/{id}'.get.security[0].SecurityScheme[0]"));

        // TODO: https://github.com/quarkusio/quarkus/issues/30997
        // assertEquals("user", json.getString("paths.'/secured/roles-allowed/{id}'.delete.security[0].SecurityScheme[0]"));
```
The last one (commented due to a bug in `smallrye-openapi`) requires an additional endpoint in https://github.com/quarkus-qe/quarkus-test-suite/blob/main/spring/spring-data/src/main/java/io/quarkus/ts/spring/data/rest/secured/RolesAllowedRepository.java:
```java
    @Override
    @RolesAllowed("user")
    void deleteById(Long aLong);
```